pandas/tests/tools/test_to_datetime.py

```diff
@@ -3628,19 +3578,18 @@ def test_to_datetime_mixed_awareness_mixed_types(aware_val, naive_val, naive_fir
             to_datetime(vec, utc=True)
 
     else:
-        with tm.assert_produces_warning(FutureWarning, match=depr_msg):
+        msg = "cannot parse datetimes with mixed time zones unless `utc=True`"
+        with pytest.raises(ValueError, match=msg):
             to_datetime(vec)
 
         # No warning/error with utc=True
         to_datetime(vec, utc=True)
 
     if both_strs:
-        with tm.assert_produces_warning(FutureWarning, match=depr_msg):
+        msg = "cannot parse datetimes with mixed time zones unless `utc=True`"
+        with pytest.raises(ValueError, match=msg):
             to_datetime(vec, format="mixed")
-        with tm.assert_produces_warning(FutureWarning, match=depr_msg):
-            msg = "DatetimeIndex has mixed timezones"
-            with pytest.raises(TypeError, match=msg):
-                DatetimeIndex(vec)
+            DatetimeIndex(vec)
```

Comment:
if they're both meant to raise, can we do this in two context managers please? else maybe only one of them could raise the error and the test would erroneously pass

instead of
```
with pytest.raises(ValueError, match=msg):
    to_datetime(vec, format="mixed")
    DatetimeIndex(vec)
```
let's do
```
with pytest.raises(ValueError, match=msg):
    to_datetime(vec, format="mixed")
with pytest.raises(ValueError, match=msg):
    DatetimeIndex(vec)
```

Reply:
thanks, I replaced one context manager with two.

I checked, both `to_datetime(vec, format="mixed")` and `DatetimeIndex(vec)` raise `ValueError: cannot parse datetimes with mixed time zones unless 'utc=True'`